src/main/java/hudson/plugins/jira/JiraRestService.java

```diff
@@ -238,22 +241,33 @@ public BasicIssue createIssue(String projectKey, String description, String assi
 
     public User getUser(String username) {
         try {
-            return jiraRestClient.getUserClient().getUser(username).get(10, TimeUnit.SECONDS);
+            return jiraRestClient.getUserClient().getUser(username).get(timeout, TimeUnit.SECONDS);
         } catch (Exception e) {
             LOGGER.warning("jira rest client get user error. cause: " + e.getMessage());
             return null;
         }
     }
-
+ 
     public void updateIssue(String issueKey, List<Version> fixVersions) {
         final IssueInput issueInput = new IssueInputBuilder().setFixVersions(fixVersions)
                                                              .build();
         try {
-            jiraRestClient.getIssueClient().updateIssue(issueKey, issueInput).get(10, TimeUnit.SECONDS);
+            jiraRestClient.getIssueClient().updateIssue(issueKey, issueInput).get(timeout, TimeUnit.SECONDS);
         } catch (Exception e) {
             LOGGER.warning("jira rest client update issue error. cause: " + e.getMessage());
         }
     }
+    
+    public void setIssueLabels(String issueKey, List<String> labels) {
+        final IssueInput issueInput = new IssueInputBuilder()
+        		.setFieldValue(IssueFieldId.LABELS_FIELD.id, labels)
+                .build();
+        try {
+            jiraRestClient.getIssueClient().updateIssue(issueKey, issueInput).get(timeout, TimeUnit.SECONDS);
+        } catch (Exception e) {
+            LOGGER.warning("jira rest client update issue error. cause: " + e.getMessage());
```

Comment:
ok, i will change it, but it will be different than rest of the class, maybe better to create separate PR in future ti fix all logging lines?